src/frontend/dash/circlePacker.ts

```diff
@@ -1,203 +1,351 @@
-
-/*
-    Margin : spacing between the circles
-    Aspect ratio : the target width-length proportions to pack the circles in
-    delta average : acts like a middle layer to size averaging and takes the change in 
-    size (averagedSize - oldSize) and multiplies it by delta average and adds it to old size 
-    to get newSize 
-
-    Size averaging : change the size of a circle to the average of itself and all previous
-    smaller circles to make them approach a similar sizes.
-*/
-let aspectRatio = 1;
-let deltaAvg = 0.95;
+import {getArrivals, busSigns} from "./trimet"
 
-let placedCircles: Circle[] = []
-let maxX;
-let minX;
-let maxY;
-let minY;
+const membersDiv = document.getElementById('members');
 
-let targetMaxX;
-let targetMaxY;
-let targetMinX;
-let targetMinY;
+export class Vector2D {
+    x : number;
+    y : number;
 
-export function getBounds() {
-    return {
-        maxX,
-        maxY,
-        minX,
-        minY
+    constructor(x = 0, y = 0) {
+        this.x = x;
+        this.y = y;
+    }
+
+    getDistanceFrom(vector : Vector2D) {
+        return Math.sqrt((this.x - vector.x)**2 + (this.y - vector.y)**2);
+    }
+
+    normalized() {
+        const divider = new Vector2D().getDistanceFrom(this);
+
+        return new Vector2D(this.x / divider, this.y / divider);
+    }
+
+    scaled(scalar : number) {
+        return new Vector2D(this.x * scalar, this.y * scalar);
+    }
+
+    added(vector : Vector2D) {
+        return new Vector2D(this.x + vector.x, this.y + vector.y);
     }
 }
 
-export class Circle {
-    x: number;
-    y: number;
+export abstract class Circle {
+    position : Vector2D;
+    velocity : Vector2D;
+    acceleration : Vector2D;
+    
     r: number;
+
+    readonly element : HTMLElement;
+
     constructor(r) {
+        this.element = membersDiv.appendChild(document.createElement("name"));
+
         this.r = r;
+        this.position = new Vector2D();
+        this.acceleration = new Vector2D();
+        this.velocity = new Vector2D();
     }
-}
-export class MemberCircle extends Circle {
-    name: string;
-    imgurl: string;
-    constructor(hours, name, imgurl) {
-        super((Math.sqrt(hours + .2)) * 10);
-        this.name = name;
-        this.imgurl = imgurl;
+
+    get mass() {
+        return Math.PI * this.r**2;
+        // return 200;
     }
-}
-export class ClockCircle extends Circle{
 
-}
+    get charge() {
+        return this.r**2 / 5;
+    }
 
-export function setAspectRatio(ratio : number) {
-    aspectRatio = ratio;
+    destroy() {
+        this.element.remove();
+        console.log(placedCircles.splice(placedCircles.indexOf(this), 0));
+    }
+
+    abstract updateSize();
 }
 
-function getDistanceFrom(circle1, x, y) {
-    return Math.sqrt(Math.pow(circle1.x - x, 2) + Math.pow(circle1.y - y, 2));
+export class MemberCircle extends Circle {
+    loginTime : number;
+    name : string;
+
+    constructor(loginTime, name, imgurl) {
+        super(Math.sqrt(.2) * 10);
```

Comment:
This was something I added in Circle Packer 2.0. 

The whole equation is meant to be 
`Math.sqrt(hours + .2) * 10`
but since hours is always 0 when the constructor is called, I just removed it.

And the slope of a square gets closer to 0 as x increases, so larger circles will be closer in size.